### PR TITLE
DOC: fix formatting in the ExtensionArray docstrings

### LINKED ISSUE #28685: DOC: Fix formatting errors in pandas.api.extensions.ExtensionArray
```python
python scripts/validate_docstrings.py pandas.api.extensions.ExtensionArray.view

python scripts/validate_docstrings.py pandas.api.extensions.ExtensionArray.fillna

python scripts/validate_docstrings.py pandas.api.extensions.ExtensionArray.shift


```
#### Problem description

pandas.api.extensions.ExtensionArray.view: Parameter "dtype" description should finish with "."
pandas.api.extensions.ExtensionArray.view: Return value has no description

pandas.api.extensions.ExtensionArray.fillna: Parameter "method" description should finish with "."
pandas.api.extensions.ExtensionArray.fillna: The first line of the returns section should contain only the type, unless multiple values are being returned
pandas.api.extensions.ExtensionArray.fillna: Return value has no description

pandas.api.extensions.ExtensionArray.shift: Parameter "fill_value" description should finish with "."
pandas.api.extensions.ExtensionArray.shift: The first line of the returns section should contain only the type, unless multiple values are being returned
pandas.api.extensions.ExtensionArray.shift: Return value has no description
```

This is a subset of #27977  and #28602

### PR DESCRIPTION
**Pull request by HughKelley**
- [x] closes #28685 
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by HughKelley** [comment]
@datapythonista, seen above, there was an existing style error in the file modified for this pull request. Should I just go ahead and fix the line that is too long?

File: pandas/core/arrays/base.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ class ExtensionArray:
     """
     Abstract base class for custom 1-D array types.
 
-    pandas will recognize instances of this class as proper arrays
+    pandas will recognize instances of this claspandas.api.extensions.ExtensionArray.views as proper arrays
```
Comment (datapythonista) [inline pandas/core/arrays/base.py]: besides the line too long, looks like something went wrong here, we need ```:class:`pandas.api...```

File: pandas/core/arrays/base.py [hunk 2]
```diff
@@ -869,11 +871,12 @@ def view(self, dtype=None) -> Union[ABCExtensionArray, np.ndarray]:
         Parameters
         ----------
         dtype : str, np.dtype, or ExtensionDtype, optional
-            Default None
+            Default None.
 
         Returns
         -------
         ExtensionArray
+            The array calling .view.
```
Comment (datapythonista) [inline pandas/core/arrays/base.py]: Can you rephrase this. And if `.view` is a method, please use Sphinx ```:meth:`...```

**Review by datapythonista — COMMENTED** [review]
Thanks for working on this, just couple of comments.

File: pandas/core/arrays/base.py (hunk 2 above)
Comment (HughKelley) [reply pandas/core/arrays/base.py]: Updated, this method just returns the ExtensionArray. The argument dtype isn't used yet. Not sure exctly what the docstring should be until this is fully implemented.

File: pandas/core/arrays/base.py (hunk 1 above)
Comment (HughKelley) [reply pandas/core/arrays/base.py]: This was just a control-f gone wrong. It's removed below.

File: pandas/core/arrays/base.py [hunk 3]
```diff
@@ -539,13 +540,14 @@ def shift(self, periods: int = 1, fill_value: object = None) -> ABCExtensionArra
...
             The scalar value to use for newly introduced missing values.
-            The default is ``self.dtype.na_value``
+            The default is ``self.dtype.na_value``.
 
             .. versionadded:: 0.24.0
 
         Returns
         -------
-        shifted : ExtensionArray
+            ExtensionArray
```
Comment (WillAyd) [inline pandas/core/arrays/base.py]: I think this needs to be unindented to render properly

File: pandas/core/arrays/base.py [hunk 4]
```diff
@@ -539,13 +540,14 @@ def shift(self, periods: int = 1, fill_value: object = None) -> ABCExtensionArra
...
+                Shifted.
```
Comment (WillAyd) [inline pandas/core/arrays/base.py]: Can remove this line altogether

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/core/arrays/base.py (hunk 4 above)
Comment (HughKelley) [reply pandas/core/arrays/base.py]: That was intended to resolve the error: `pandas.api.extensions.ExtensionArray.shift: Return value has no description`

File: pandas/core/arrays/base.py (hunk 3 above)
Comment (HughKelley) [reply pandas/core/arrays/base.py]: agreed, thanks

Comment (datapythonista) [reply pandas/core/arrays/base.py]: @HughKelley can you fix this please, so we can merge. Thanks!

Comment (HughKelley) [reply pandas/core/arrays/base.py]: @datapythonista should I make the change with an additional commit or is there another way to do this?

Comment (TomAugspurger) [reply pandas/core/arrays/base.py]: Additional commit.

File: pandas/core/arrays/base.py [hunk 5]
```diff
@@ -869,11 +871,12 @@ def view(self, dtype=None) -> Union[ABCExtensionArray, np.ndarray]:
         Parameters
         ----------
         dtype : str, np.dtype, or ExtensionDtype, optional
-            Default None
+            Default None.
 
         Returns
         -------
         ExtensionArray
+            A view of the :class:`ExtensionArray` calling :meth:`view`.
```
Comment (TomAugspurger) [inline pandas/core/arrays/base.py]: I don’t think the :meth: resolves to anything.

Comment (HughKelley) [reply pandas/core/arrays/base.py]: thanks, It looked to me as though that functionality was not fully built out so I just tried to describe it as literally as possible in order to pass the checks. Would ```A view of the :class:`ExtensionArray```` be sufficient?

Comment (datapythonista) [reply pandas/core/arrays/base.py]: yes, I think removing the `calling...` is good enough, thanks @HughKelley

File: pandas/core/arrays/base.py (hunk 3 above)
Comment (datapythonista) [reply pandas/core/arrays/base.py]: Can you fix the indantation here @HughKelley please?

**Review by datapythonista — APPROVED** [review]
lgtm, thanks @HughKelley

**Comment by HughKelley** [comment]
> 
> 
> lgtm, thanks @HughKelley

First contribution to a project. Thanks for the help and direction!

**Comment by datapythonista** [comment]
> First contribution to a project. Thanks for the help and direction!

You're welcome. Hopefully not the last one. :)

**Review by WillAyd — APPROVED** [review]
lgtm as well ping when CI is green

**Comment by HughKelley** [comment]
@WillAyd thanks!

**Comment by WillAyd** [comment]
Thanks @HughKelley
